From 9f66ef7b73af9ae9628f7a1eb0008e01b7d5054b Mon Sep 17 00:00:00 2001
From: Hariharan K <quic_harihk@quicinc.com>
Date: Tue, 23 Jan 2024 15:32:42 +0530
Subject: [PATCH] pcie: qcom: Fix KW issue in pci probe

The struct resource field in the pcie probe
was uninitialized before trying to ioremap
the parf register space. Avoiding this issue
by fetching the resource and getting the size.

Change-Id: I33cc61987d012641f18f17b326011bd7a5b6ce4a
Signed-off-by: Hariharan K <quic_harihk@quicinc.com>
(cherry picked from commit 760174066eca951813f578be062d02daa1cf536b)
---
 drivers/pci/controller/dwc/pcie-qcom.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/pci/controller/dwc/pcie-qcom.c b/drivers/pci/controller/dwc/pcie-qcom.c
index 64bd004288e9..efc81d4ddde2 100644
--- a/drivers/pci/controller/dwc/pcie-qcom.c
+++ b/drivers/pci/controller/dwc/pcie-qcom.c
@@ -2062,6 +2062,7 @@ static int qcom_pcie_probe(struct platform_device *pdev)
 	}
 
 	/* get the parf size which is needed for pcie_parf_read() */
+	res = platform_get_resource_byname(pdev, IORESOURCE_MEM, "parf");
 	if (res)
 		pcie->parf_size = resource_size(res);
 
-- 
2.34.1

